### ISSUE TITLE
Bug: user-defined struct types typecheck problem.

### ISSUE THREAD

**Issue by Milky2018** [Milky2018/lane#1]
Example:

```
def main : Int = 
  let p1 = Person { name = "Tom", age = 18 }
    , p2 = Person { name = "Tim", age = 20 }
  in p1.age - p2.age
```

This Lane program type checks and evaluates to -2. But according to the designing, it should not type checks.

The problem is caused by: 

```haskell
tc (EStruct n fields) Nothing env = do
  fieldTypes <- mapM fieldType fields
  return $ TVStruct n fieldTypes where
    fieldType (field, expr) = do
      t <- tc expr Nothing env
      return (field, t)
```

**Comment by Milky2018** [comment]
Thanks! We will fix this ASAP.

**Comment by Milky2018** [comment]
Fixed by commit [edf9f47](https://github.com/Milky2018/lane/commit/edf9f47fd79e4299c2ed9d3dd68e86816cb1707b).
